aidial_adapter_openai/legacy_completions.py

```diff
@@ -0,0 +1,87 @@
+from typing import Any, Dict
+
+from aidial_sdk.exceptions import HTTPException as DialException
+from fastapi.responses import JSONResponse, StreamingResponse
+from openai import AsyncStream
+from openai.types import Completion
+
+from aidial_adapter_openai.env import COMPLETION_DEPLOYMENTS_PROMPT_TEMPLATES
+from aidial_adapter_openai.utils.auth import OpenAICreds
+from aidial_adapter_openai.utils.parsers import (
+    AzureOpenAIEndpoint,
+    OpenAIEndpoint,
+)
+from aidial_adapter_openai.utils.reflection import call_with_extra_body
+from aidial_adapter_openai.utils.sse_stream import to_openai_sse_stream
+from aidial_adapter_openai.utils.streaming import build_chunk, map_stream
+
+
+def convert_to_chat_completions_response(
+    chunk: Completion, is_stream: bool
+) -> Dict[str, Any]:
+    return build_chunk(
+        id=chunk.id,
+        finish_reason=chunk.choices[0].finish_reason,
+        delta=chunk.choices[0].text,
+        created=str(chunk.created),
+        is_stream=is_stream,
+        usage=chunk.usage.to_dict() if chunk.usage else None,
+    )
+
+
+async def chat_completion(
+    data: Dict[str, Any],
+    endpoint: OpenAIEndpoint | AzureOpenAIEndpoint,
+    creds: OpenAICreds,
+    api_version: str,
+    is_stream: bool,
+    deployment_id: str,
+) -> Any:
+
+    if data.get("n", 1) > 1:  # type: ignore
+        raise DialException(
+            status_code=422,
+            message="The deployment doesn't support n > 1",
+            type="invalid_request_error",
+        )
+
+    client = endpoint.get_client({**creds, "api_version": api_version})
+
+    messages = data.get("messages", [])
+
+    prompt = messages[-1].get("content") or ""
+
+    if (
+        template := COMPLETION_DEPLOYMENTS_PROMPT_TEMPLATES.get(deployment_id)
+    ) is not None:
+        prompt = template.format(prompt=prompt)
+
+    if len(messages) == 0:
```

Comment:
Move this check before `messages[-1]` is accessed.

Reply:
Fixed